ddtrace/contrib/pymongo/client.py

```diff
@@ -285,3 +307,11 @@ def _set_query_metadata(span, cmd):
         span.resource = "{} {} {}".format(cmd.name, cmd.coll, q)
     else:
         span.resource = "{} {}".format(cmd.name, cmd.coll)
+
+
+def set_query_rowcount(span, cursor):
+    search_key = "Batch"
+    # results returned in batches, get len of each batch
+    rowcount = sum([len(documents) for batch_key, documents in cursor.items() if search_key in batch_key])
```

Comment:
Iterating over the cursor will pop data entries from the cursor object: https://github.com/mongodb/mongo-python-driver/blob/3.5.1/pymongo/cursor.py#L1138, this will consume the cursor iterable and the tracer will return an empty cursor object to users. We should avoid adding this side effect (I am not 100% sure about this, we can confirm this hypothesis by adding a test case). If this is actually the case we can avoid this issue by using `cursor.clone().items()`.

Reply:
I thought this as well but haven't noticed of these side affects, atleast within Python (PHP cursors behaved as expected and iterating resulted in missing documents). For most of this Pymongo client code, the "cursor" object stored within the result isn't an actual cursor, just a plain dictionary.

Reply:
Also, we already have tests that iterate through our result cursor and check that the length is as expected.